File: 7-reverse-integer/reverse-integer.java
class Solution {
     public int reverse(int x) {
        double ans = 0;
        
        while (x != 0) {
            if (ans * 10 > Integer.MAX_VALUE || ans * 10 < Integer.MIN_VALUE) return 0;
            ans *= 10;
            ans += x % 10;
            x /= 10;
            
        }
        return (int) ans;
    }
}

File: 7-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int :
        neg: int = -1 if x < 0 else 1 
        
        x = abs(x)
        res = 0
        while x > 0:
            res = (res * 10) + (x % 10)
            if res > 2 ** 31 - 1:
                return 0
            x //= 10
        return res * neg
        